typed-racket-lib/typed-racket/base-env/signature-prims.rkt

```diff
@@ -0,0 +1,101 @@
+#lang racket/base
+
+;; This file implements unit signatures for typed racket
+
+(provide define-signature)
+
+(require "../utils/utils.rkt"
+         "colon.rkt"
+         (for-syntax syntax/parse
+                     racket/base
+                     racket/list
+                     racket/syntax
+                     syntax/kerncase
+                     "../private/syntax-properties.rkt"
+                     "../typecheck/internal-forms.rkt"
+                     syntax/id-table
+                     racket/dict
+                     racket/unit-exptime
+                     (utils tc-utils))
+         (only-in racket/unit
+                  [define-signature untyped-define-signature]
+                  extends)
+         (for-label "colon.rkt")
+         (submod "../typecheck/internal-forms.rkt" forms)
+         (only-in "../../typed/racket/base.rkt" define-type)
+         (for-template (rep type-rep)))
+
+(begin-for-syntax 
+  (define-literal-set colon #:for-label (:))
+  
+  ;; TODO: there should be a more extensible way of handling signatures
+  (define-syntax-class signature-forms
+    (pattern (form:def-sig-form ...)))
+  
+  (define-syntax-class def-sig-form
+    #:attributes (internal-form erased)
+    (pattern :sig-var-form
+             #:attr kind 'var)
+    
+    (pattern :sig-type-form
+             #:attr kind 'type))
+  
+  (define-syntax-class sig-var-form
+    #:literal-sets (colon)
+    (pattern [name:id : type]
+             #:with internal-form #'(name type)
+             #:with erased #'name))
+
+  ;; Preliminary support for type definitions in signatures
+  ;; The form is allowed in signature definitions, but currently
+  ;; is ignored
```

Comment:
I don't remember if I commented on this previously, but can we just disable them (i.e., not parse them) until we support them?